File: src/main/scala/net/machinemuse/numina/gui/NuminaGuiHandler.scala
package net.machinemuse.numina.gui

import cpw.mods.fml.common.network.IGuiHandler
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.world.World
import cpw.mods.fml.relauncher.{Side, SideOnly}
import net.minecraft.client.Minecraft
import net.minecraft.stats.AchievementList
import net.machinemuse.numina.death.GuiGameOverPlus
import net.minecraft.inventory.ContainerPlayer

/**
 * Author: MachineMuse (Claire Semple)
 * Created: 7:59 PM, 10/15/13
 */
object NuminaGuiHandler extends IGuiHandler {
  override def getServerGuiElement(ID: Int, player: EntityPlayer, world: World, x: Int, y: Int, z: Int): AnyRef = {
    ID match {
      case _ => null
    }
  }

  @SideOnly(Side.CLIENT)
  override def getClientGuiElement(ID: Int, player: EntityPlayer, world: World, x: Int, y: Int, z: Int): AnyRef = {
    Minecraft.getMinecraft.thePlayer.addStat(AchievementList.openInventory, 1)
    ID match {
      case 0 => new GuiGameOverPlus()
      case _ => null
    }
  }
}

File: src/main/java/net/machinemuse/numina/gui/NuminaGuiHandler.java
package net.machinemuse.numina.gui;

import cpw.mods.fml.common.network.IGuiHandler;
import cpw.mods.fml.relauncher.Side;
import cpw.mods.fml.relauncher.SideOnly;
import net.machinemuse.numina.death.GuiGameOverPlus;
import net.minecraft.client.Minecraft;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.stats.AchievementList;
import net.minecraft.world.World;

/**
 * Author: MachineMuse (Claire Semple)
 * Created: 7:59 PM, 10/15/13
 *
 * Ported to Java by lehjr on 10/10/16.
 */
public class NuminaGuiHandler implements IGuiHandler {
    @Override
    public Object getServerGuiElement(int ID, EntityPlayer player, World world, int x, int y, int z) {
        return null;
    }

    @Override
    @SideOnly(Side.CLIENT)
    public Object getClientGuiElement(int ID, EntityPlayer player, World world, int x, int y, int z) {
        Minecraft.getMinecraft().thePlayer.addStat(AchievementList.openInventory, 1);

        if (ID== 0)
            return new GuiGameOverPlus();
        return null;
    }
}
